### PR TITLE
Migrate `PluginDetailActivity` to ViewBinding

### PR DESCRIPTION
**Pull request by neeldoshii**
## Description 

This pull request migrates the `PluginDetailActivity`  to use ViewBinding, improving type safety and eliminating the need for findViewById calls. 

## Fixes 
* #19180

## Screenshots/ Video

// TODO

## Steps to reproduce 
1. Ensure that you are on `Wasabi` build variant.
2. Click on my Site
3. In manage --> Click on Plugins
4. Click on any plugin
5. Play with the plugin :D

## PR Submission Checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java [hunk 1]
```diff
@@ -865,7 +808,7 @@ private void cancelDomainCreditsCheckProgressDialog() {
             mCheckingDomainCreditsProgressDialog.cancel();
         }
     }
-
+    @SuppressWarnings("deprecation") // Progress Dialog Deprecation on API 26
```
Comment (neeldoshii) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Reviewer Note : 

Do we have any issue which says of migrating the deprecated Progress Dialog to ProgressBar/Notification? If so then should I comment issue number over here for ease in understanding the issue?

Comment (neeldoshii) [reply WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Reverting this due to not related to context of the PR.

**Comment by neeldoshii** [comment]
CC @ParaskP7 request to review once you are free.

File: WordPress/src/main/res/layout/plugin_detail_activity.xml [hunk 2]
```diff
@@ -433,7 +433,9 @@
                 </LinearLayout>
             </com.google.android.material.card.MaterialCardView>
 
-            <include layout="@layout/plugin_ratings_cardview" />
+            <include
+                android:id="@+id/plugin_ratings_cardview"
```
Comment (ParaskP7) [inline WordPress/src/main/res/layout/plugin_detail_activity.xml:437]: Heart (❤️): Nicely done! 🥇

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java [hunk 3]
```diff
@@ -1285,6 +1228,7 @@ private void startAutomatedTransfer() {
      * updates site settings etc, it'll be lost when the Automated Transfer is completed. The process takes about 1 min
      * on average, and we'll be able to update the progress by checking the status of the transfer.
      */
+
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Minor (🔍): Consider removing this empty line.

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java [hunk 4]
```diff
@@ -1421,7 +1365,7 @@ public void onAutomatedTransferInitiated(OnAutomatedTransferInitiated event) {
      * Unfortunately we can't close the progress dialog until both the site and its plugins are fetched. Check out
      * `onSiteChanged` for the next step.
      */
-    @SuppressWarnings("unused")
+    @SuppressWarnings("unused") // Progress Dialog Deprecation on API 26
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Minor (🔍): Consider removing this extra comment here ([context](https://github.com/wordpress-mobile/WordPress-Android/pull/20968#discussion_r1637463943)).

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java [hunk 5]
```diff
@@ -805,47 +747,48 @@ private void confirmRemovePlugin() {
...
 
     private void showPluginRemoveFailedSnackbar() {
-        WPSnackbar.make(mContainer,
+        WPSnackbar.make(mBinding.pluginDetailContainer,
                 getString(R.string.plugin_remove_failed, mPlugin.getDisplayName()),
                 Snackbar.LENGTH_LONG)
                   .show();
     }
 
+
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Minor (🔍): Consider removing this empty line.

**Review by ParaskP7 — APPROVED** [review]
👋 @neeldoshii !

Thank you so much for your contribution to JP/WPAndroid with this PR! 🥇 

I have reviewed and tested this PR as per the instructions, along with triggering CI on it via a [draft](https://github.com/wordpress-mobile/WordPress-Android/pull/21009) PR, which I just pushed on the main repo, everything works as expected, good job! 🌟 

-----

I have left few minor (🔍) comments for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java [hunk 6]
```diff
@@ -390,6 +355,7 @@ protected void onDestroy() {
 
         mDispatcher.unregister(this);
         super.onDestroy();
+        mBinding = null;
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Warn (⚠️): Just something I forgot to mention here, when using ViewBinding on the activity level, it is not necessary to nullify the binding during `onDestroy()`. This is only necessary on the fragment level, see docs, [activities](https://developer.android.com/topic/libraries/view-binding#activities) vs [fragments](https://developer.android.com/topic/libraries/view-binding#fragments):

```
Note: Fragments outlive their views. Make sure you clean up any references to the binding
class instance in the fragment's onDestroyView() method.
```

Comment (neeldoshii) [reply WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Thanks for the above docs explanation, did the changes!

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java (hunk 4 above)
Comment (neeldoshii) [reply WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Done!

**Comment by neeldoshii** [comment]
Thank you for reviewing @ParaskP7 🚀.  I have applied the requested changes! 😅

> I am NOT going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

I don't think I have the permission to merge the PR. 😅

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java (hunk 6 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java]: Nice, thanks Neel! 👍

**Comment by ParaskP7** [comment]
Thanks for applying my suggestions @neeldoshii ! 🥇 

> I don't think I have the permission to merge the PR. 😅

Yea, contributors don't have the permission to merge, but also, I first need to re-trigger CI on the changes you made. Let me do this work and merge this if everything goes well! 👍

**Comment by ParaskP7** [comment]
🎉 @neeldoshii , another contribution in, thanks so much for all your work! 🥇

**Pull request merged**